routing/osrm_path_segment_factory.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "routing/loaded_path_segment.hpp"
+
+#include "std/unique_ptr.hpp"
+
+namespace routing
+{
+struct RoutingMapping;
+struct RawPathData;
+struct FeatureGraphNode;
+
+namespace turns
+{
+using TSeg = OsrmMappingTypes::FtSeg;
+
+// General constructor.
+unique_ptr<LoadedPathSegment> LoadedPathSegmentFactory(RoutingMapping & mapping,
```

Comment:
> bool LoadPathSegment(..., LoadedPathSegment & segment);

It's a possible solution. But to implement that we need:
1. to create LoadedPathSegment object before LoadPathSegment call. For the time being LoadPathSegment does not have a default constructor. Pls have a look at constructing LoadedPathSegment objet in the factories.

2. to have more code for processing the LoadPathSegment result.